### PR TITLE
DEV-2329: Identify Alerting requirements for Assignment Client reliable operation

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Jira: https://highfidelity.atlassian.net/browse/DEV-2329

Decorated some of the logging with the PAGE: and ALERT: keywords for the start-up of generic part of assignment clients.

Domain server PR: https://github.com/highfidelity/hifi/pull/16285

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21782/HighFidelity-Beta-PR16309-f73a7c0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21782/HighFidelity-Quest-Beta-PR16309-f73a7c0.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21782/HighFidelity-Beta-PR16309-f73a7c0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21782/HighFidelity-Beta-Interface-PR16309-f73a7c0.dmg

**Comment by SimonWalton-HiFi** [comment]
build this please

File: assignment-client/src/AssignmentFactory.cpp [hunk 1]
```diff
@@ -24,7 +24,9 @@
 ThreadedAssignment* AssignmentFactory::unpackAssignment(ReceivedMessage& message) {
 
     quint8 packedType;
-    message.peekPrimitive(&packedType);
+    if (message.readPrimitive(&packedType) != sizeof(packedType)) {
```
Comment (howard-stearns) [inline assignment-client/src/AssignmentFactory.cpp]: Do none of the various assignment classes (in the switch below) do anything with message? It seems like changing this from peek to read would require a change to any that do, no?

File: assignment-client/src/AssignmentClient.cpp [hunk 2]
```diff
@@ -183,16 +186,21 @@ void AssignmentClient::sendAssignmentRequest() {
             // we want to check again for the local domain-server port in case the DS has restarted
             quint16 localAssignmentServerPort;
             if (nodeList->getLocalServerPortFromSharedMemory(DOMAIN_SERVER_LOCAL_PORT_SMEM_KEY, localAssignmentServerPort)) {
-                if (localAssignmentServerPort != _assignmentServerSocket.getPort()) {
-                    qCDebug(assignment_client) << "Port for local assignment server read from shared memory is"
-                        << localAssignmentServerPort;
-
-                    _assignmentServerSocket.setPort(localAssignmentServerPort);
-                    nodeList->setAssignmentServerSocket(_assignmentServerSocket);
+                if (localAssignmentServerPort == 0) {
```
Comment (howard-stearns) [inline assignment-client/src/AssignmentClient.cpp]: nice

File: assignment-client/src/AssignmentFactory.cpp (hunk 1 above)
Comment (SimonWalton-HiFi) [reply assignment-client/src/AssignmentFactory.cpp]: Thanks - it looks like they might use it in some cases. I'll change it back to peek.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21788/HighFidelity-Beta-PR16309-f73a7c0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21788/HighFidelity-Quest-Beta-PR16309-f73a7c0.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21791/HighFidelity-Beta-PR16309-f63360a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21791/HighFidelity-Quest-Beta-PR16309-f63360a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21791/HighFidelity-Beta-PR16309-f63360a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21791/HighFidelity-Beta-Interface-PR16309-f63360a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21791/HighFidelity-Beta-PR16309-f63360a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21791/HighFidelity-Beta-Interface-PR16309-f63360a.exe

**Pull request merged**